assembly/index.ts

```diff
@@ -26,110 +40,93 @@ function FACTOR(length: i32): i32 {
 export function encode(source: Uint8Array): string {
 
   // Skip & count leading zeroes.
-  let zeroes = 0
-  let length = 0
-  let pbegin = 0
-  let pend = source.length
-
-  while (pbegin != pend && source[pbegin] == 0) {
-    pbegin++
-    zeroes++
-  }
+  let pend = source.length;
+  let pbegin = 0;
+  while (pbegin != pend && source[pbegin] == 0) ++pbegin;
+  let zeroes = pbegin;
 
   // Allocate enough space in big-endian base58 representation.
-  let size = ((pend - pbegin) * iFACTOR + 1) >>> 0
-  let b58 = new Uint8Array(size)
+  let size = INV_FACTOR(pend - pbegin);
+  let b58 = new Uint8Array(size);
+  let length = 0;
 
   // Process the bytes.
   while (pbegin != pend) {
-    let carry = i32(source[pbegin])
-
+    let carry = u32(source[pbegin])
     // Apply "b58 = b58 * 256 + ch".
     let i = 0
-    for (let it = size - 1; (carry != 0 || i < length) && (it != -1); it--, i++) {
-      carry += (256 * b58[it]) >>> 0
-      b58[it] = (carry % BASE) >>> 0
-      carry = (carry / BASE) >>> 0
+    for (let it = size - 1; it != -1 && (carry != 0 || i < length); --it, ++i) {
+      carry += u32(b58[it]) << 8;
+      b58[it] = carry % BASE;
+      carry = carry / BASE;
     }
-
-    assert(carry == 0, 'Non-zero carry');
-    length = i
-    pbegin++
+    if (ASC_OPTIMIZE_LEVEL == 0) {
```

Comment:
Why this branch?

Reply:
Could `Non-zero carry` happened with fixed alphabet set?

Reply:
up

Reply:
I think it's a debug check.  I think this is good to go.